exp/rendering-architecture.md

```diff
@@ -0,0 +1,54 @@
+
+For NVIDIA GPUs, Anbox Cloud driver stacks currently have a native OpenGL driver. For AMD and Intel GPUs, an OpenGL or EGL driver is layered on top of the Vulkan API. Since Vulkan API provides better GPU management at a lower level than OpenGL or EGL, this approach is beneficial and preferred by many users.
+
+To have a better understanding of the rendering architecture of Anbox Cloud, it is important to understand what the Android framework offers in terms of rendering. In Android, applications interact with the SurfaceFlinger which is the system compositor that is responsible for composing a frame together from all the outputs rendered by different applications. The frame is then submitted to the hardware composer which renders the frames on a screen. For more information on Android graphics components and how they work, see https://source.android.com/docs/core/graphics. 
+
+# Rendering pipeline
+
+Anbox Cloud has two rendering pipeline models - one for NVIDIA and the other for Intel and AMD. However, irrespective of the GPU that you use, the path of a frame typically looks like this: Android application > SurfaceFlinger > Hardware composer > Anbox Cloud > Display on screen or send it to streaming component.
+
+For communication between the hardware composer module on the Android side and Anbox runtime, we use [Wayland](https://wayland.freedesktop.org/). So Anbox Cloud really functions as a compositor for Android i.e., the hardware composer module receives frames from the SurfaceFlinger and notifies Anbox runtime using Wayland. The Anbox runtime then submits the frame towards its output, which is either the screen or the streaming component.
+
+## For NVIDIA
+
+![Anbox Cloud NVIDIA pipeline|690x440](https://assets.ubuntu.com/v1/3ba1fddd-NVIDIA_pipeline.png)
+
+This NVIDIA pipeline uses the EmuGL stack. EmuGL is the hardware GLES emulation that is implemented with a mix of translator libraries, system libraries, and other rendering components. To understand more about EmuGL, see [EmuGL design overview](https://android.googlesource.com/platform/external/qemu/+/6654d90be28d0059baf854caccd4caf9f7033ccb/android/android-emugl/DESIGN). 
+
+For NVIDIA, we cannot use the NVIDIA driver inside the Android container because of compatibility issues. Hence, we use the Enterprise Ready NVIDIA driver that is available on every Ubuntu installation. So by using the EmuGL stack, we have an Anbox Cloud GPU driver which is a standard OpenGL ES or EGL driver that receives the API calls and converts them to remote procedure calls to the NVIDIA driver. The actual rendering and actions on the NVIDIA driver happens on the Anbox runtime side inside the Ubuntu container and not in the Android space.
+
+In terms of performance, compared to how rendering is done on Intel or AMD GPUs using Vulkan on top of ANGLE, using the EmuGL stack is suboptimal as it leads to unnecessary overhead due to transmission of GL calls from the Android space to Anbox runtime.
```

Comment:
Can we word this a bit lighter in the sense of "... leads to very minimal additional overhead". Right now it sounds like its making things performing really bad which it by no means does. It's performant and and serves the need of cloud gaming with no compromise. Only certain operations are going to slower than with the direct approach (e.g. `glReadPixels`, but this one is slow anyway).